src/client/api/v1/get_node_info.c

```diff
@@ -6,9 +6,63 @@
 #include "client/network/http.h"
 #include "core/utils/iota_str.h"
 
+res_node_info_t *res_node_info_new() {
+  res_node_info_t *res = malloc(sizeof(res_node_info_t));
+  if (res == NULL) {
+    printf("[%s:%d]: OOM\n", __func__, __LINE__);
+    return NULL;
+  }
+  res->is_error = false;
+  return res;
+}
+
+void res_node_info_free(res_node_info_t *res) {
+  if (res) {
+    if (res->is_error) {
+      res_err_free(res->u.error);
+    } else {
+      if (res->u.output_node_info->features) {
+        utarray_free(res->u.output_node_info->features);
+      }
+      free(res->u.output_node_info);
+    }
+    free(res);
+  }
+}
+
+char *get_node_features_at(res_node_info_t *info, size_t idx) {
+  if (info == NULL) {
+    printf("[%s:%d]: get_features failed (null parameter)\n", __func__, __LINE__);
+    return NULL;
+  }
+
+  int len = utarray_len(info->u.output_node_info->features);
+  if (idx < 0 || idx >= len) {
+    printf("[%s:%d]: get_features failed (invalid index)\n", __func__, __LINE__);
+    return NULL;
+  }
+
+  return *(const char **)utarray_eltptr(info->u.output_node_info->features, idx);
+}
+
+size_t get_node_features_num(res_node_info_t *info) {
+  if (info == NULL) {
+    printf("[%s:%d]: get_features failed (null parameter)\n", __func__, __LINE__);
+    return NULL;
```

Comment:
```
../src/client/api/v1/get_node_info.c:51:12: warning: incompatible pointer to integer conversion returning 'void *' from a function with result type 'size_t' (aka 'unsigned long') [-Wint-conversion]
    return NULL;
           ^~~~
```

Reply:
could you clarify what you meant [here](https://github.com/iotaledger/iota.c/pull/78#discussion_r551138678)?

It was unclear to me what you meant, so I assumed you confused *bracket* with *parenthesis*.
I removed the extra parenthesis, but it generated the following:

[`warning: using the result of an assignment as a condition without parentheses`](https://github.com/iotaledger/iota.c/runs/1646073872#step:6:952)

Reply:
sorry, you are right, it's parentheses.